osu.Game/Screens/Edit/Compose/Components/SelectionHandler.cs

```diff
@@ -455,6 +469,12 @@ public MenuItem[] ContextMenuItems
         {
             get
             {
+                if (shiftPressed)
+                {
+                    deleteSelected();
+                    return null;
+                }
```

Comment:
Hijacking the context menu to delete the selected object like this instead is an incredibly workaround-y way of doing this. You'd probably be looking to do this in `HandleSelectionRequested` instead:

```csharp
if (state.Keyboard.ShiftPressed && state.Mouse.IsPressed(MouseButton.Right))
{
    EditorBeatmap.Remove(blueprint.HitObject);
    return;
}
```

although that method is getting a little bit complex already with that addition.